pkgs/development/compilers/llvm/common/clang-tools/default.nix

```diff
@@ -10,50 +13,99 @@
   enableLibcxx ? false,
 }:
 
-stdenv.mkDerivation {
-  unwrapped = clang-unwrapped;
+let
+  clang-tools = stdenv.mkDerivation {
+    # clang-unwrapped extracts headers outside the main derivation, into
+    # clang-unwrapped.lib - this is at odds with clangd's header auto-detection
+    # logic, which assumes it'll have them at <prefix>/lib.
+    #
+    # For simplicity, instead of working around this with the `CPATH` and
+    # `CPLUS_INCLUDE_PATH` envs, let's just reconstruct the directory structure
+    # expected by clangd - this avoids extra logic within the wrapper script.
```

Comment:
Unless I'm misunderstanding, this comment is outright misleading. It suggests the `CPATH` / `CPLUS_INCLUDE_PATH` workaround is a viable solution and the reconstruction of the directory structure is just out of simplicity, while the reality is the `CPATH` / `CPLUS_INCLUDE_PATH` break as many things as they fix and are not a suitable solution at all.